### PR TITLE
CLOUD-755 Demand image everywhere inside CR

### PR DESCRIPTION
**Pull request by cap1984**
[CLOUD-755](https://jira.percona.com/browse/CLOUD-755) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ type PerconaServerMongoDBSpec struct {
 	ClusterServiceDNSSuffix string                               `json:"clusterServiceDNSSuffix,omitempty"`
 	ClusterServiceDNSMode   DnsMode                              `json:"clusterServiceDNSMode,omitempty"`
 	Sharding                Sharding                             `json:"sharding,omitempty"`
-	InitImage               string                               `json:"initImage,omitempty"`
+	InitImage               string                               `json:"initImage"`
```
[the remark this answers is not in the document]
Comment (cap1984) [reply pkg/apis/psmdb/v1/psmdb_types.go]: Removed it according to https://github.com/percona/percona-server-mysql-operator/pull/260

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/init-deploy.log) |
| upgrade-consistency | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/upgrade-consistency.log) |
| one-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/one-pod.log) |
| storage | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/storage.log) |
| cross-site-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/cross-site-sharded.log) |
| limits | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/limits.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/demand-backup.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/monitoring-2-0.log) |
| self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/self-healing-chaos.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/scaling.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/scheduled-backup.log) |
| operator-self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/operator-self-healing-chaos.log) |
| security-context | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/security-context.log) |
| demand-backup-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/demand-backup-sharded.log) |
| arbiter | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/arbiter.log) |
| rs-shard-migration | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/rs-shard-migration.log) |
| upgrade | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/upgrade.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/service-per-pod.log) |
| liveness | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/liveness.log) |
| upgrade-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/upgrade-sharded.log) |
| smart-update | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/smart-update.log) |
| pitr | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/pitr.log) |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/version-service.log) |
| pitr-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/pitr-sharded.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/0265915a/users.log) |
| We run 25 out of 28|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1102/commits/0265915a2fb3d11d1e4f9749b0e0da6195095831
image: `perconalab/percona-server-mongodb-operator:PR-1102-0265915a`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/storage.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/one-pod.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/init-deploy.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/upgrade-consistency.log) |
| cross-site-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/cross-site-sharded.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/demand-backup.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/self-healing-chaos.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/monitoring-2-0.log) |
| limits | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/limits.log) |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/arbiter.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/scaling.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/operator-self-healing-chaos.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/scheduled-backup.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/security-context.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/service-per-pod.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/rs-shard-migration.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/liveness.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/smart-update.log) |
| demand-backup-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/demand-backup-sharded.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/version-service.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/users.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/data-sharded.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/non-voting.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/upgrade.log) |
| demand-backup-eks-credentials | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/demand-backup-eks-credentials.log) |
| data-at-rest-encryption | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/data-at-rest-encryption.log) |
| upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/upgrade-sharded.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/pitr.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/f106178c/pitr-sharded.log) |
| We run 29 out of 28|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1102/commits/f106178cf38ca39b2b72b5389909965ccf3813b4
image: `perconalab/percona-server-mongodb-operator:PR-1102-f106178c`

**Review by tplavcic — APPROVED** [review]

**Review by inelpandzic — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/storage.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/one-pod.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/upgrade-consistency.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/init-deploy.log) |
| cross-site-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/cross-site-sharded.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/demand-backup.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/self-healing-chaos.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/monitoring-2-0.log) |
| limits | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/limits.log) |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/arbiter.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/scaling.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/operator-self-healing-chaos.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/scheduled-backup.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/security-context.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/service-per-pod.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/rs-shard-migration.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/liveness.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/smart-update.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/version-service.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/users.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/data-sharded.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/non-voting.log) |
| demand-backup-eks-credentials | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/demand-backup-eks-credentials.log) |
| data-at-rest-encryption | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/data-at-rest-encryption.log) |
| demand-backup-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1102/6b734fb9/demand-backup-sharded.log) |
| We run 25 out of 28|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1102/commits/6b734fb964f5fa105162a9c4964fba9f273acafc
image: `perconalab/percona-server-mongodb-operator:PR-1102-6b734fb9`

**Pull request merged**